cartographer/mapping_2d/sparse_pose_graph.cc

```diff
@@ -114,10 +116,14 @@ void SparsePoseGraph::AddScan(
     submap_states_.back().trajectory = submaps;
     CHECK_EQ(submap_states_.size(), submap_indices_.size());
   }
+  // ... One call before the first of insertion_submaps was finished
```

Comment:
I'm not sure I understand this comment. We check here, whether this was the last insertion into the submap. This is the case when `finished_probability_grid` is non-null.